src/System.Net.HttpListener/src/System/Net/HttpListenerResponse.cs

```diff
@@ -61,22 +61,15 @@ public void SetCookie(Cookie cookie)
             {
                 throw new ArgumentNullException(nameof(cookie));
             }
-            bool added = false;
-            try
-            {
-                Cookies.Add(cookie);
-                added = true;
-            }
-            catch (CookieException)
```

Comment:
Desktop does not catch anything. In fact I'm fairly sure this `CookieException` catching is dead code. From referencesource:

```
public void Add(Cookie cookie) {
  if (cookie == null) {
      throw new ArgumentNullException("cookie");
  }
  ++m_version;
  int idx = IndexOf(cookie);
  if (idx == -1) {
      m_list.Add(cookie);
  }
  else {
      m_list[idx] = cookie;
  }
}

internal int IndexOf(Cookie cookie) {
    IComparer comp = Cookie.GetComparer();
    int idx = 0;
    foreach (Cookie c in m_list) {
        if (comp.Compare(cookie, c) == 0) {
            return idx;
        }
        ++idx;
    }
    return -1;
}
```

Either way we have to use `CookieCollection.InternalAdd` as it behaves differently for cookies of different variants than `CookieCollection.Add`